social/strategies/django_strategy.py

```diff
@@ -63,7 +62,14 @@ def request_path(self):
 
     def request_port(self):
         """Port in use for this request"""
-        return self.request.META['SERVER_PORT']
+        try:  # django >= 1.9
+            return self.request.get_port()
+        except AttributeError:  # django < 1.9
+            host_parts = self.request.get_host().split(':')
+            try:
+                return host_parts[1]
+            except IndexError:
+                return None
```

Comment:
We should only hit this code path if the port is 80 or 443, in which case it will be stripped anyway: https://github.com/onelogin/python-saml/blob/efb2515e0a12fb43271d5d6b8bddcbaa4a61f451/src/onelogin/saml2/utils.py#L268-L271